src/js/flags/getFlagForName.ts

```diff
@@ -0,0 +1,56 @@
+import {debugModeFlags} from './debug-mode';
```

Comment:
camelCase filenames is inconsistent with the remainder of the project.  I see why you don't want to put this in flag.ts, since it would create what _looks_ like a circular dependency (though at runtime it's not).  Assuming we add prefix to FlagSection (or even just store them in a dict, keyed by the prefix - that could work too), then this dict is already going to live in flagset.ts.  Does this seem reasonable as a static function (or separate top-level function) on FlagSet?

Reply:
I was trying to avoid having the separate dictionary since it means maintaining two copies of the data (one in the flag section and the dictionary itself). Your call, this was a function designed to traverse the flag sections for the correct flag in order to find the conflicts (also stored on the flag now). I don't see an issue moving to flag set; I moved it out to try and keep the size of the flag set file down for readability.

Reply:
Forgot to ask, for the filenames, `_` or `-` for word breaks?

Reply:
You've already reduced flagset substantially.  Where's the extra copy?  You obviously need a copy for the definition, and a copy that imports it, but that's it:

```
// someflags.ts
export const debugMode = {...};
```

```
// flagset.ts
import {debugMode} from './flags/debugmode.js';
import {easyMode} from './flags/easymode.js';

const sections = [debugMode, easyMode];
const sectionsByPrefix = new Map(sections.map(s => [s.prefix, s]));
```

I think we can get away with only listing them all out once, no?

Reply:
The extra copy of some of the info would've come from keeping the flag conflicts map around (which is why I replaced it with that set of lookup functions in the flagset class).